aspnetcore/blazor/components.md

```diff
@@ -536,6 +536,18 @@ protected override void OnAfterRender()
 }
 ```
 
+### Complete asynchronous actions before rendering
+
+Asynchronous actions performed in lifecycle events may not have completed before the component is rendered. Objects might be `null` or incompletely populated with data while the lifecycle method is executing. Provide rendering logic to confirm that objects are initialized. Render placeholder UI elements (for example, a loading message) while objects are `null`.
+
+In the Fetch Data component of the Blazor templates, `OnInitAsync` is overridden to asychronously receive forecast data (`forecasts`). When `forecasts` is `null`, a loading message is displayed to the user. After the `Task` returned by `OnInitAsync` completes, the component is rerendered with the updated state.
+
+*Pages/FetchData.razor*:
+
+[!code-cshtml[](components/samples_snapshot/3.x/FetchData.razor?highlight=9-18,21-26)]
```

Comment:
Are you sure you're highlighting the right lines here? I would have thought the emphasis would be on the if statement to show how you need to have this check in your code.

Reply:
> would be on the if statement

The entire `@if` block and the content of the `@code` was highlighted because the paragraph speaks to all of this code. However if the main point of the whole section is the conditional, sure thing ... let's just highlight the `@if` on Line 9. Will get it on the next commit.